flow360/component/simulation/translator/solver_translator.py

```diff
@@ -0,0 +1,180 @@
+import re
+from typing import Union
+
+from flow360.component.simulation.models.surface_models import (
+    Freestream,
+    SlipWall,
+    SymmetryPlane,
+    Wall,
+)
+from flow360.component.simulation.models.volume_models import Fluid
+from flow360.component.simulation.outputs.outputs import (
+    SliceOutput,
+    SurfaceOutput,
+    TimeAverageVolumeOutput,
+    VolumeOutput,
+)
+from flow360.component.simulation.simulation_params import SimulationParams
+from flow360.component.simulation.translator.utils import (
+    get_attribute_from_first_instance,
+    preprocess_input,
+)
+from flow360.component.simulation.unit_system import LengthType
+
+
+def to_dict(input_params):
+    return input_params.model_dump(by_alias=True, exclude_none=True)
+
+
+def to_dict_without_unit(input_params):
+    unit_keys = {"value", "units"}
+    output_dict = to_dict(input_params)
+    for key, value in output_dict.items():
+        output_dict[key] = (
+            value["value"] if isinstance(value, dict) and value.keys() == unit_keys else value
+        )
+    return output_dict
+
+
+def remove_empty_keys(input_dict):
+    # TODO: implement
```

Comment:
get_attribute_from_first_instance can return None. I don't want to do the check before setting an entry every time, so I thought we can have a function to remove keys for None values when translation is done

Reply:
So I think in theory get_attribute_from_first_instance should never return None since we work on preprocessed param. Otherwise we are missing required information.

Reply:
It's possible for some fields to be optional right?

Reply:
They can be optional but should have been populated by the time they reach translator. But let's see as we go.